_ml-commons-plugin/agents-tools/tools/connector-tool.md

```diff
@@ -0,0 +1,168 @@
+---
+layout: default
+title: Connector tool
+has_children: false
+has_toc: false
+nav_order: 20
+parent: Tools
+grand_parent: Agents and tools
+---
+
+<!-- vale off -->
+# Connector tool
+**Introduced 2.15**
+{: .label .label-purple }
+<!-- vale on -->
+
+The `ConnectorTool` uses a [connector]({{site.url}}{{site.baseurl}}/ml-commons-plugin/remote-models/connectors/) to call any REST API function. For example, you can use a `ConnectorTool` to call a Lambda function through its REST API interface.
+
+## Step 1: Register a connector with an execute action
+
+The `ConnectorTool` can only run an `execute` action within a connector. Before you can create a `ConnectorTool`, configure a connector and provide an `execute` action in the `actions` array. The `execute` action is used to invoke a function at a REST API endpoint. It is similar to the `predict` action, which is used to invoke a machine learning (ML) model. 
+
+For this example, you'll create a connector for a simple AWS Lambda function that accepts two integers and returns their sum. This function is hosted on a dedicated endpoint with a specific URL, which you'll provide in the `url` parameter. For more information, see [Lambda function URLs](https://docs.aws.amazon.com/lambda/latest/dg/lambda-urls.html).
+
+To create a connector, send the following request:
+
+```json
+POST _plugins/_ml/connectors/_create
+{
+  "name": "Lambda connector of simple calculator",
+  "description": "Demo connector of lambda function",
+  "version": 1,
+  "protocol": "aws_sigv4",
+  "parameters": {
+    "region": "YOUR AWS REGION",
+    "service_name": "lambda"
+  },
+  "credential": {
+    "access_key": "YOUR ACCESS KEY",
+    "secret_key": "YOUR SECRET KEY",
+    "session_token": "YOUR SESSION TOKEN"
+  },
+  "actions": [
+    {
+      "action_type": "execute",
+      "method": "POST",
+      "url": "YOUR LAMBDA FUNCTION URL",
+      "headers": {
+        "content-type": "application/json"
+      },
+      "request_body": "{ \"number1\":\"${parameters.number1}\", \"number2\":\"${parameters.number2}\" }"
+    }
+  ]
+}
+```
+{% include copy-curl.html %} 
+
+OpenSearch responds with a connector ID:
+
+```json
+{
+  "connector_id": "Zz1XEJABXWrLmr4mewEF"
+}
+```
+
+## Step 2: Register a flow agent that will run the ConnectorTool
+
+For this example, the Lambda function adds the two input numbers and returns their sum in the `result` field:
+
+```json
+{
+  "result": 5
+}
+```
+
+By default, the `ConnectorTool` expects the response from the Lambda function to contain a field named `response`. However, in this example the Lambda function response doesn't include a `response` field. To retrieve the result from the `result` field instead, you need to provide a `response_filter`, specifying the dot path to the `result` field (`$.result`). Using the `response_filter`, the `ConnectorTool` will retrieve the result from the `result` field of the Lambda function response and return it in the OpenSearch response.
```

Comment:
dot path should be json path, also add this link https://github.com/json-path/JsonPath

Reply:
```suggestion
By default, the `ConnectorTool` expects the response from the Lambda function to contain a field named `response`. However, in this example the Lambda function response doesn't include a `response` field. To retrieve the result from the `result` field instead, you need to provide a `response_filter`, specifying the [json path](https://github.com/json-path/JsonPath) to the `result` field (`$.result`). Using the `response_filter`, the `ConnectorTool` will retrieve the result with json path and return filtered result in `response`.
```